frontend/lib/resolution/return-type-inference.cpp

```diff
@@ -889,7 +889,8 @@ static bool helpComputeReturnType(Context* context,
     }
 
     // if there are no returns with a value, use void return type
-    if (fnAstReturnsNonVoid(context, ast->id()) == false) {
+    if (fn->linkage() != Decl::EXTERN &&
+        fnAstReturnsNonVoid(context, ast->id()) == false) {
```

Comment:
Does this mean an `extern proc`  with no type specified will not be void? Should it be?

Reply:
I _think_ this is in the part of return type inference that requires a look at the function body, which ``extern`` does not have. If the return type is specified in the signature, then I don't think this part of the code will run.